src/main/java/slimeknights/mantle/command/BookCommand.java

```diff
@@ -0,0 +1,183 @@
+import net.minecraft.commands.Commands;
+import net.minecraft.commands.arguments.ResourceLocationArgument;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.player.Player;
+import org.joml.Matrix4f;
+import slimeknights.mantle.Mantle;
+import slimeknights.mantle.client.book.BookLoader;
+import slimeknights.mantle.client.book.data.BookData;
+import slimeknights.mantle.client.screen.book.BookScreen;
+import slimeknights.mantle.network.packet.OpenNamedBookPacket;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/** A command for different book */
+public class BookCommand {
+  private static final String EXPORT_SUCCESS = "command.mantle.book.export.success";
+  private static final String EXPORT_FAIL = "command.mantle.book.export.error_generic";
+  private static final String EXPORT_FAIL_IO = "command.mantle.book.export.error_io";
+
+  /**
+   * Registers this sub command with the root command
+   * @param subCommand  Command builder
+   */
+  public static void register(LiteralArgumentBuilder<CommandSourceStack> subCommand) {
+    subCommand.requires(source -> source.hasPermission(MantleCommand.PERMISSION_GAME_COMMANDS) && source.getEntity() instanceof AbstractClientPlayer)
+      .then(Commands.literal("open")
+        .then(Commands.argument("id", ResourceLocationArgument.id()).suggests(MantleClientCommand.REGISTERED_BOOKS)
+          .executes(BookCommand::openBook)))
+      .then(Commands.literal("export_images")
+        .then(Commands.argument("id", ResourceLocationArgument.id()).suggests(MantleClientCommand.REGISTERED_BOOKS)
+          .executes(BookCommand::exportImages)));
+  }
+
+  /**
+   * Opens the specified book
+   * @param context  Command context
+   * @return  Integer return
+   */
+  private static int openBook(CommandContext<CommandSourceStack> context) {
+    ResourceLocation book = ResourceLocationArgument.getId(context, "id");
+
+    BookData bookData = BookLoader.getBook(book);
+    if(bookData != null) {
+      bookData.openGui(Component.literal("Book"), "", null, null);
+    } else {
+      OpenNamedBookPacket.ClientOnly.errorStatus(book);
+      return 1;
+    }
+
+    return 0;
+  }
+
+  /**
+   * Renders all images in the book to files
+   * @param context  Command context
+   * @return  Integer return
+   */
+  private static int exportImages(CommandContext<CommandSourceStack> context) {
+    ResourceLocation book = ResourceLocationArgument.getId(context, "id");
+
+    BookData bookData = BookLoader.getBook(book);
+
+    Path gameDirectory = Minecraft.getInstance().gameDirectory.toPath();
+    Path screenshotDir = Paths.get(gameDirectory.toString(), Screenshot.SCREENSHOT_DIR, "mantle_book", book.getNamespace(), book.getPath());
+
+    if(bookData != null) {
+      int width = BookScreen.PAGE_WIDTH_UNSCALED * 2;
+      int height = BookScreen.PAGE_HEIGHT_UNSCALED;
```

Comment:
I am curious, what GUI scale does this export under? I recall there is a min scale for the HD font we use in tinkers books to work, plus some things like item stacks look better at a larger scale.

Might be worth an optional argument to specify the GUI scale just in case a larger one is needed.

Reply:
Did a quick test in game, seems they ignore the GUI scale setting, which is a good way to do it. However, I do believe the export will not work with Tinkers' Construct's usage of an HD font as the scale is just a little too small.

So we probably want a scale parameter, default it to 1 but let it be set higher in case books use such elements.